theano/sandbox/multinomial.py

```diff
@@ -149,18 +174,24 @@ def perform(self, node, ins, outs):
         nb_outcomes = pvals.shape[1]
 
         # For each multinomial, loop over each possible outcome
-        for n in range(nb_multi):
-            waiting = True
-            cummul = 0
-            unis_n = unis[n]
-
-            for m in range(nb_outcomes):
-                cummul += pvals[n, m]
-                if (waiting and (cummul > unis_n)):
-                    z[0][n, m] = 1
-                    waiting = False
-                else:
-                    z[0][n, m] = 0
+        for c in range(n_samples):
+            for n in range(nb_multi):
+                waiting = True
+                cummul = 0
+                unis_n = unis[n]
```

Comment:
@carriepl This is a bug. It should be unis[c*nb_multi + n]. Should I create a new pull request, or can we re-open this one?